cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -226,6 +226,28 @@ gdv_int32 round_int32(gdv_int32 num) { return num; }
 FORCE_INLINE
 gdv_int64 round_int64(gdv_int64 num) { return num; }
 
+#define FACTORIAL(IN_TYPE)                                                          \
+  FORCE_INLINE                                                                      \
+  gdv_int64 factorial_##IN_TYPE(gdv_int64 ctx, gdv_##IN_TYPE value) {               \
```

Comment:
since only limited number of inputs are valid just use a look up table

Reply:
I applied your suggestions, check if it is fine.